src/components/forms/Combobox.tsx

```diff
@@ -780,15 +794,20 @@ export function ComboboxAsync<T = any>({
   }, [inputOptions.value]);
 
   useEffect(() => {
-    if (enableQuery && inputOptions.value && !disableWithQueryParameter) {
+    if (
+      enableQuery &&
+      inputOptions.value &&
+      !disableWithQueryParameter &&
+      !isEntryAvailable()
+    ) {
       $url.searchParams.set('with', inputOptions.value.toString());
     }
-  }, [enableQuery]);
+  }, [enableQuery, inputOptions.value]);
 
   const { data } = useQuery(
     [
-      new URL(url).pathname,
-      new URL(url).searchParams.toString(),
+      new URL($url).pathname,
+      new URL($url).searchParams.toString(),
```

Comment:
@beganovich This is exactly what we discussed in today's meeting. We need to include `$url` instead of the default `url` to call the API, otherwise, React Query will not be able to determine when the API needs to be called. I noticed this during this task and made the change.